assets/js/translator.js

```diff
@@ -188,6 +217,90 @@ if(modeEnabled('translation')) {
             $('a#fileDownload').fadeOut('fast');
         });
 
+        $('#translatedText').css('height', $('#originalText').css('height'));
+        $('#suggestCloseBtn').click(function() {
+            $('#suggestedWordInput').val('');
+            grecaptcha.reset();
+        });
+        $('#suggestBtn').click(function() {
+            var fromWord = $('#suggestionTargetWord').html();
+            var toWord = $('#suggestedWordInput').val();
+            var recaptchaResponse = grecaptcha.getResponse();
+
+            if(toWord.length === 0) {
+                $('#suggestedWordInput').tooltip('destroy');
+                $('#suggestedWordInput').tooltip({
+                    'title': 'Suggestion cannot be empty.',
+                    'trigger': 'manual',
+                    'placement': 'bottom'
+                });
+                $('#suggestedWordInput').tooltip('show')
+                setTimeout(function() {
+                    $('#suggestedWordInput').tooltip('hide');
+                    $('#suggestedWordInput').tooltip('destroy');
+                }, 3000);
```

Comment:
Whaooo. What are we doing here? :\ We should _never_ use timeouts unless they're absolutely absolutely necessary. There should be an event that you can find in the bootstrap docs that you can hook on to that will let you do this. Probably, [`hidden.bs.tooltip`](http://getbootstrap.com/javascript/#tooltips-methods).